src/arch/wasm/CodeGen.zig

```diff
@@ -3289,6 +3290,11 @@ fn airNot(func: *CodeGen, inst: Air.Inst.Index) InnerError!void {
     func.finishAir(inst, result, &.{ty_op.operand});
 }
 
+fn airTrap(func: *CodeGen, inst: Air.Inst.Index) InnerError!void {
+    try func.addTag(.@"unreachable");
+    func.finishAir(inst, .none, &.{});
+}
+
 fn airBreakpoint(func: *CodeGen, inst: Air.Inst.Index) InnerError!void {
```

Comment:
@Luukdegram if we look at LLVM and see what `llvm.trap` and `llvm.debugtrap` compile down to, (https://zig.godbolt.org/z/dvqzYT6M5), you can find that they both compile down to Wasm's `unreachable` instruction (for the lack of anything better I'm assuming). I'm wondering if, although we don't support DWARF for Wasm yet, it would be an improvement over status quo to emit `unreachable` in `airBreakpoint` too? I don't know what "itfunc" is BTW in case that has anything to do with it.

Reply:
Yes, trap and breakpoint should both lower to the wasm opcode "unreachable". I was just looking at this yesterday. This will make the os.abort() implementation better and make wasmtime actually print a stack trace on panic.